lib/rescue_from_duplicate/active_record/extension.rb

```diff
@@ -62,12 +67,13 @@ def postgresql_exception_columns(exception)
     end
 
     def sqlite3_exception_columns(exception)
-      extract_columns(exception.message[/columns? (.*) (?:is|are) not unique/, 1])
+      extract_columns(exception.message[/columns? (.*) (?:is|are) not unique/, 1]) || 
+      extract_columns(exception.message[/UNIQUE constraint failed: ([^:]*)\:/, 1])
     end
 
     def extract_columns(columns_string)
       return unless columns_string
-      columns_string.split(",").map(&:strip)
+      columns_string.split(",").map { |column| column.split('.').last.strip }
```

Comment:
Right. This is because in that new format, columns are in their "fully qualified names", e.g. `table_name.column_name`, not just `column_name`